### ISSUE TITLE
bug on writeCoilToBuffer

### ISSUE THREAD

**Issue by madjarian** [yaacov/ArduinoModbusSlaveTCP#2]
i think i found a bug in writeCoilToBuffer:

void ModbusTCP::writeCoilToBuffer(int offset, uint16_t state) {
    int address = MLEN + 3 + offset / 8;
    int bit = offset % 8;

    if (state == HIGH) {
        bitSet(bufOut[address], bit);
    } else if (state) {                                 <-- this is wrong ! and must be (!state)
        bitClear(bufOut[address], bit);
    }
}
are you ok ?

**Comment by yaacov** [comment]
Thanks 👍  

Can you check if changing the line to just simple `} else { ` in
https://github.com/yaacov/ArduinoModbusSlaveTCP/blob/master/src/ModbusSlaveTCP.cpp#L245
works ?

If it does, can you make a pull request to fix it ?

**Comment by madjarian** [comment]
Hi Yaacov
i made test with writecoil on esp8266 , with  it didnt work.
```
if (state == HIGH) {
    bitSet(bufOut[address], bit);
} else if (state) {  
    bitClear(bufOut[address], bit);
}
```

with that it's ok !
```
if (state == HIGH) {
    bitSet(bufOut[address], bit);
} else if (!state) {  
    bitClear(bufOut[address], bit);
}
```

**Comment by yaacov** [comment]
fixed by #3
